PSDepend/Public/Get-Dependency.ps1

```diff
@@ -395,12 +411,28 @@ function Get-Dependency {
                     DependsOn = Get-GlobalOption -Name DependsOn -Prefer $DependencyHash.DependsOn
                     PreScripts = Get-GlobalOption -Name PreScripts -Prefer $DependencyHash.PreScripts
                     PostScripts = Get-GlobalOption -Name PostScripts -Prefer $DependencyHash.PostScripts
-                    PSDependOptions = $PSDependOptions
+					Credential = Resolve-Credential -Name $CredentialName
+					PSDependOptions = $PSDependOptions
                     Raw = $DependencyHash
                 }
             }
         }
-    }
+	}
+
+	# Heleper to retrieve the credential for a dependency
+	function Resolve-Credential  {
+		[CmdletBinding()]
+		param (
+			[string]$Name
+		)
+
+		$credential = $null
+		if (($null -ne $Name) -and ($null -ne $Credentials)) {
+			$Credential = $Credentials[$Name]
```

Comment:
It might be worth using $Credentials.ContainsKey($Name) in the logic just to avoid exceptions in cases where someone misidentifies a credential?  Or at the very least, to catch that case and write a useful error or warning?

Reply:
I agree. Let me know if the latest changes address your concern.